src/tiledb/cloud/bioimg/ingestion.py

```diff
@@ -68,13 +70,30 @@ def ingest(
     :param output_ext: extension for the output images in tiledb
     :param tile_scale: The scaling factor applied to each tile during I/O.
         Larger scale factors will result in less I/O operations.
-    :param access_credentials_name: Access Credentials Name (ACN) registered
-        in TileDB Cloud (ARN type)
+    :param access_credentials_name: [TBDeprecated] Access Credentials Name (ACN)
+        registered in TileDB Cloud (ARN type) if `acn` is not set.
     """
 
     logger = get_logger_wrapper(verbose)
     max_workers = None if num_batches else 20  # Default picked heuristically.
 
+    # Demand for mutual exclusion of the two arguments and existence.
+    access_credentials_name = kwargs.pop("access_credentials_name", None)
+    if not (bool(acn) != bool(access_credentials_name)):
+        raise ValueError(
+            "Ingestion graph requires `access_credentials_name`"
+            "or `acn` mutually exclusively to be set."
+        )
+    # Backwards compatibility: Assign when only access_credentials_name is set
+    if not acn:
+        acn = access_credentials_name
+        warnings.warn(
+            "The 'access_credentials_name' parameter is about to be"
+            "deprecated and will be removed in future versions."
+            "Please use the 'acn' parameter instead.",
+            DeprecationWarning,
```

Comment:
I'm a fan of using the normal class-style syntax for building warnings instead:

```python
warnings.warn(
    DeprecationWarning(
        "The blah blah blah"
    )
)
```

though this isn’t a huge deal.